File: src/python/Q9095_Adding123.py
def Adding(n):
  if n == 1: return 1
  elif n == 2: return 2
  elif n == 3: return 4
  else: return Adding(n - 1) + Adding(n - 2) + Adding(n - 3)
 
for i in range(int(input())):
  num = int(input())
  print(Adding(num))

File: src/java/backjoon/Q9095_Adding123.java
package backjoon;

import java.util.Scanner;

public class Q9095_Adding123 {
	public static void main(String[] args) {
		int[] Adding = new int[12];
		Adding[1] = 1;
		Adding[2] = 2;
		Adding[3] = 4;
		
		for (int i = 4; i < 12; i++) {
			Adding[i] = Adding[i - 1] + Adding[i - 2] + Adding[i - 3];
		}
		
		Scanner scan = new Scanner(System.in);
		byte n = scan.nextByte();
		while (n-- > 0) {
			byte num = scan.nextByte();
			System.out.println(Adding[num]);
		}
		scan.close();
	}
}
